ADR/0039-send-cloud-events.md

```diff
@@ -0,0 +1,34 @@
+# 40. Konflux should send cloud events all system events.
+
+Date: 2024-09-24
+
+## Status
+
+Proposed
+
+## Context
+
+Konflux had made the architectural decision to not use cloud events. However, that does not mean that
+Konflux should not emit cloud events.
+
+Emitting cloud events would allow Konflux users to easily track what is happening in the system. In addition,
+they can use these cloud events to create their own product-specific infrastructure to support their build
+and release process.
+
+To support this, all Konflux components should be required to emit cloud events for signicant events. These
+should be documented fully and made available for users.
+
+Cloud event generation could be optional and that option could default to off. But users should be able to
```

Comment:
This probably needs clarification on what "users" are. I'm assuming this is a Konflux deployment admin, and not a user that uses Konflux to build content.

Reply:
From async conversation - I think Greg means for this to be end-users. I.e., the developers that use Konflux to build, test and release their software. He doesn't mean platform engineers (the people running a konflux instance) or other konflux controllers.